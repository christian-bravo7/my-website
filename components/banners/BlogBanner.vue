<template>
  <div class="container mx-auto py-20">
    <div class="flex flex-col items-center | mb-12">
      <h1 class="text-7xl font-bold | mb-2">
        {{ $t(greetingKey) }}
      </h1>
      <p class="text-lg">
        {{ $t('blog.welcome-text') }}
      </p>
    </div>
    <div class="flex justify-center">
      <AppInput
        id="post-search"
        v-model="value"
      />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator';

import { greetingByHour } from '@/utils/date/greetingByHour';

@Component
export default class BlogBanner extends Vue {
  value = '';

  get greetingKey (): string {
    const greetingKey = greetingByHour();

    return `greeting.${greetingKey}`;
  }
}
</script>
