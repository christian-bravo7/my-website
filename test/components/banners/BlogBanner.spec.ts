import { mount } from '@vue/test-utils';

import BlogBanner from '@/components/banners/BlogBanner.vue';

describe('BlogBanner component', () => {
  it('should render correclty', () => {
    const wrapper = mount(BlogBanner, {
      propsData: {
        image: 'image.png',
        date: [2020, 10, 24],
        title: 'my new post',
        description: 'welcome to my new post',
        minutes: 2,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should format created date', () => {
    const wrapper = mount(BlogBanner, {
      propsData: {
        image: 'image.png',
        date: [2020, 10, 24],
        title: 'my new post',
        description: 'welcome to my new post',
        minutes: 2,
      },
    });

    const date = wrapper.find('span.text-xs.text-center.text-gray-300');

    expect(date.text()).toBe('Oct 24, 2020');
  });
});
