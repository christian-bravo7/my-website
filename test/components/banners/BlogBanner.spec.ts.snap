// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogBanner component should render correclty 1`] = `
<section class="flex relative bg-black"><img src="image.png" alt="ballon" class="absolute inset-0 w-full h-full object-cover object-center opacity-50">
  <div class="flex flex-col justify-between container mx-auto relative z-10 p-4">
    <div></div>
    <div class="flex flex-col justify-center items-center"><span class="text-xs text-center text-gray-300">
        Oct 24, 2020
      </span>
      <h1 class="text-3xl text-center md:text-5xl text-gray-50 font-bold">
        my new post
      </h1>
      <h6 class="text-base  text-center text-gray-50">
        welcome to my new post
      </h6>
    </div>
    <div class="flex justify-end"><span class="text-gray-50 text-xs"><span class="text-xl">
          2
        </span>
      min read
      </span></div>
  </div>
</section>
`;
